src/test/java/it/pagopa/transactions/client/PaymentGatewayClientTest.java

```diff
@@ -132,8 +136,12 @@ void shouldReturnAuthorizationResponseCP_XPAY() throws JsonProcessingException {
                 .thenReturn(Mono.just(xPayResponse));
 
         /* test */
-        StepVerifier.create(client.requestGeneralAuthorization(authorizationData))
-                .assertNext(response -> response.getT2().get().equals(xPayResponse))
+        StepVerifier.create(client.requestXPayAuthorization(authorizationData))
+                .assertNext(response -> response.equals(xPayResponse))
```

Comment:
If you use `expectNext` you get nicer console output (also watch out, `assertNext` expects you to throw assertion errors inside the lambda body, `expectNext` does that for you)

```suggestion
                .expectNext(xPayResponse)
```